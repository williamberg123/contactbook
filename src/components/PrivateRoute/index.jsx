import PropTypes from 'prop-types';

export default function PrivateRoute({ children, isAuth }) {
	if (!isAuth) {
		window.location.href = '/';
	} else {
		return children;
	}
}

PrivateRoute.propTypes = {
	children: PropTypes.node.isRequired,
	isAuth: PropTypes.bool.isRequired,
};
