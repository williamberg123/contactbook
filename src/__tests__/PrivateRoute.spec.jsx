import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import PrivateRoute from '../components/PrivateRoute';

describe('<PrivateRoute />', () => {
	it('should render the children element', () => {
		render(
			<PrivateRoute isAuth>
				<h1>Rota renderizada</h1>
			</PrivateRoute>,
		);

		expect(screen.getByRole('heading', { name: /rota renderizada/i })).toBeInTheDocument();
	});

	it('should not render the children element and redirect route', () => {
		render(
			<PrivateRoute isAuth={false}>
				<h1>Rota renderizada</h1>
			</PrivateRoute>,
		);

		expect(screen.queryByRole('heading', { name: /rota renderizada/i })).not.toBeInTheDocument();
		expect(window.location.href).toBe('http://localhost:3000/');
	});

	it('should match snapshot', () => {
		render(
			<PrivateRoute isAuth>
				<h1>Rota renderizada</h1>
			</PrivateRoute>,
		);

		expect(screen.queryByRole('heading', { name: /rota renderizada/i })).toMatchSnapshot();
	});
});
