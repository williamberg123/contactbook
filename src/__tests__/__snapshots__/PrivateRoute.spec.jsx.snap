// Vitest Snapshot v1

exports[`<PrivateRoute /> > should match snapshot 1`] = `
<h1>
  Rota renderizada
</h1>
`;
